src/fhir/api.ts

```diff
@@ -128,3 +128,27 @@ const createEntry = (resource: Resource, identifier?: string) => {
     request: requestInfo,
   };
 };
+
+export const getErrors = (r: any, resourceTypes?: string[]) => {
+  //should be result of response.json()
+  const errorArray: ErrorDetails[] = [];
+  const errorResponses = [];
+  let count = 0;
+  for (const entry of r.entry) {
+    if (!entry.response.status.toString().startsWith("2")) {
+      entry.count = count;
+      errorResponses.push(entry);
+    }
+    count++;
+  }
+  errorResponses.forEach((error) => {
+    const errorDetails: ErrorDetails = {
+      errorCode: error.response.status,
+      resourceType: resourceTypes?.[error.count] as string,
+      diagnostics: error.response.outcome.issue?.[0].diagnostics as string,
+    };
+    errorArray.push(errorDetails);
```

Comment:
I'll have a go at doing it in one but iirc there were issues looping through because the information we needed was on two branches and wouldn't recognise their relationships to each other

Reply:
I could be missing something but at a glace it looks we can build up the entire errorDetail in line 140 and push it to errorDetails